pkg/crc/preflight/preflight_checks.go

```diff
@@ -0,0 +1,325 @@
+		return false, errors.New("IP forwarding is disabled")
+	}
+	return true, nil
+}
+
+func fixIpForwardingEnabled() (bool, error) {
+	path, err := exec.LookPath("sh")
+	if err != nil {
+		return false, err
+	}
+	cmd := exec.Command(path, "-c", "echo 'net.ipv4.ip_forward = 1' | sudo tee /etc/sysctl.d/99-ipforward.conf")
+	err = cmd.Run()
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+func checkDockerMachineDriverKvmInstalled() (bool, error) {
+	// Check if docker-machine-driver-kvm is available
+	path, err := exec.LookPath("docker-machine-driver-kvm")
+	if err != nil {
+		return false, err
+	}
+	fi, _ := os.Stat(path)
+	// follow symlinks
+	if fi.Mode()&os.ModeSymlink != 0 {
+		path, err = os.Readlink(path)
+		if err != nil {
+			return false, err
+		}
+	}
+	// Check if permissions are correct
+	if fi.Mode()&0011 == 0 {
+		return false, errors.New("docker-machine-driver-kvm do not have correct permissions")
+	}
+	return true, nil
+}
+
+func fixDockerMachineDriverInstalled() (bool, error) {
+	// Download the driver binary
+	os.MkdirAll(driverBinaryDir, 0751)
+
+	out, err := os.Create(kvmDriverBinaryPath)
+	if err != nil {
+		return false, err
+	}
+	defer out.Close()
+	resp, err := http.Get(kvmDriverDownloadURL)
+	if err != nil {
+		return false, err
+	}
+	defer resp.Body.Close()
+
+	_, err = io.Copy(out, resp.Body)
+	if err != nil {
+		return false, err
+	}
+	err = out.Chown(0, 0)
+	if err != nil {
+		return false, err
+	}
+
+	err = out.Chmod(os.ModeSetuid | 0751)
+	if err != nil {
+		return false, err
+	}
+	return false, nil
+}
+
+func checkDefaultPoolAvailable() (bool, error) {
+	// Check if the default pool by libvirt is available
+	cmd := exec.Command("virsh", "--connect", "qemu:///system", "pool-list")
+	//cmd.Env = cmdUtil.ReplaceEnv(os.Environ(), "LC_ALL", "C")
+	//cmd.Env = cmdUtil.ReplaceEnv(cmd.Env, "LANG", "C")
+
+	out, err := cmd.Output()
+	if err != nil {
+		return false, nil
+	}
+
+	stdOut := string(out)
+	outputSlice := strings.Split(stdOut, "\n")
+	for _, l := range outputSlice {
+		l = strings.TrimSpace(l)
+		match, err := regexp.MatchString("^default\\s", l)
+		if err != nil {
+			return false, err
+		}
+		if match {
+			return true, nil
+		}
+	}
+	return true, nil
+}
+
+func fixDefaultPoolAvailable() (bool, error) {
+	// Create the default pool
+	return true, nil
+}
+
+func checkDefaultPoolHasSufficientSpace() (bool, error) {
+	var freeSpace poolXMLAvailableSpace
+	// Check free space in default pool
+	cmd := exec.Command("virsh", "--connect", "qemu:///system", "pool-dumpxml", "default")
+	stdOut, err := cmd.Output()
+	if err != nil {
+		return false, err
+	}
+	err = xml.Unmarshal(stdOut, &freeSpace)
+	if err != nil {
+		return false, err
+	}
+
+	if size, _ := units.FromHumanSize(defaultPoolSize); int64(freeSpace.Available) >= size {
+		return true, nil
+	}
+	return false, fmt.Errorf("Not enough space in default pool, available free space: %s", units.HumanSize(freeSpace.Available))
```

Comment:
perhaps we should actually not use the default pool. as the KVM driver uses user session (at least, the images are located in the home folder)

Reply:
Hmm... that means we could define a new pool just for crc purpose (probably have to use the system session) and then the driver can use that